TDAT: public void finish ( ) throws consequence exception {  <NL> try {  <NL> boc . resolve instances ( context ) ;  <NL> boc . create atomics ( context ) ;  <NL> java . util . vector atomics = boc . get atomics ( ) ;  <NL> do atomics ( atomics ) ;  <NL>  } catch ( consequence exception e ) {  <NL> consequence log . log . error ( " failed to finish consequence on "  <NL>  + context . get current ( ) + " : [ " + boc + " ] " + " because : "  <NL>  + e . get message ( ) ) ;  <NL>  <NL>  }  <NL>  <NL>  }  COM: <s> this methode is called by the manager when the consequence shoul be </s>